### ISSUE TITLE
Release 2.14.1 is not mirrored on packagist

### ISSUE THREAD

**Issue by Ocramius** [laminas/laminas-validator#86]
### Bug Report

|    Q        |   A
|------------ | ------
| Version(s)  | 2.14.1

#### Summary

On https://packagist.org/packages/laminas/laminas-validator, release 2.14.1 is not visible.

I forced an update, and it states: "Last update: 2021-01-22 11:38:59 UTC ".

Unsure whether the problem lies on @laminas or Packagist.

**Comment by samsonasik** [comment]
probably related with hook, looking at settings, it got:

![Screen Shot 2021-01-22 at 19 50 26](https://user-images.githubusercontent.com/459648/105492841-20a9c900-5ceb-11eb-9f47-57363178ce98.png)

"Last delivery was not succesfull. An exception occurred" https://github.com/laminas/laminas-validator/settings/hooks

**Comment by Ocramius** [comment]
Right, but that is unrelated: indeed `laminas.dev` failed to renew their certificate, but unsure how/why that would affect packagist.

On 2021-01-13 18:23:31 I see a successful push to packagist.org:

```json
{
  "ref": "refs/tags/2.14.1",
```

Response was a 202:

```json
{"status":"success","jobs":["18d86d192664d3e7f2b01ddb6dc48d72cd98159c"],"type":"github_auto"}
```

**Comment by Seldaek** [comment]
Get rid of https://github.com/laminas/laminas-validator/blob/1c7dc75bdfc0ee03e04e7200f20f1eff9b095a95/composer.json#L5 - it serves no purpose except for causing issues like this.

**Comment by Ocramius** [comment]
Oh wow, didn't even realise the `version` was there :O 

Good catch, thanks @Seldaek!

**Comment by Ocramius** [comment]
Cool, offending commit ( https://github.com/laminas/laminas-validator/commit/4e69dc696866578859bed583ea52d6e49ef04213#diff-d2ab9925cad7eac58e0ff4cc0d251a937ecf49e4b6bf57f8b95aab76648a9d34 ) reverted ( https://github.com/laminas/laminas-validator/commit/56e1353e358bd003f52511433792b30682136630#diff-d2ab9925cad7eac58e0ff4cc0d251a937ecf49e4b6bf57f8b95aab76648a9d34 ).

I'll still need to re-tag `2.14.1` as `2.14.2` after removing the version there, since it still blocks the packagist update.

**Comment by boesing** [comment]
I remember that we had to add that as PRs almost always run into circular dependency hell.
There are multiple ways to fix that:

- adding version to `composer.json` which has to be removed after CI passed (CI wont pass as it cannot install all dependencies when we are running on branches)
- adding constant `COMPOSER_ROOT_VERSION` to CI configuration to tell composer what version this currently is

I wonder if it is possible to set that `COMPOSER_ROOT_VERISON` automatically depending on the targetted branch of a PR. This would enable us to get rid of such commits which have to be removed before merging.

**Comment by Ocramius** [comment]
@boesing doing it in the PR is fine - our mistake to merge with the field still there.

**Comment by boesing** [comment]
Sure, but I wonder if it would be possible to get the target branch within GHA and parse it to provide proper informations for `COMPOSER_ROOT_VERSION` so we dont have to commit the version anymore.

Edit: seems so :-) https://stackoverflow.com/a/62332270/1328527

**Comment by Ocramius** [comment]
@boesing yes, by doing a `git checkout -b x.y.z` in the build (or calling the branch as such) where `x.y.z` is the version you want to be on.

**Comment by Ocramius** [comment]
Note: `2.14.1` will unlikely ever appear in packagist, given the mistake in releasing `2.14.0` :-)

**Comment by boesing** [comment]
Wouldn't it be possible to drop the tag and re-add it on the fixed commit which is also `2.14.2`?
As `2.14.1` and `2.14.2` dont have differences, this *might* be possible?

**Comment by Ocramius** [comment]
Possible, but not worth doing.
